flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -144,6 +144,11 @@ private static Configuration mergeHiveConf(Configuration hadoopConf, String hive
       URL configFile = CatalogLoader.class.getClassLoader().getResource("hive-site.xml");
       if (configFile != null) {
         newConf.addResource(configFile);
+      } else {
+        // If can not find hive-site.xml from classpath, then try to load resource from system env.
+        // this would be useful if don't set `hive-conf-dir` key in ddl.
+        String hiveHome = System.getenv("HIVE_HOME");
+        newConf.addResource(new Path(hiveHome + "/conf/hive-site.xml"));
```

Comment:
Could we provide an unit tests like [this](https://github.com/apache/iceberg/blob/d132474e7c8706f504a47f75b58e5a58d0bef9ff/flink/src/test/java/org/apache/iceberg/flink/TestFlinkHiveCatalog.java) ?

Reply:
Maybe we do not need a unit tests for this, since we use system environment to get `HIVE_HOME`. But we don't have a method called `System.setEnv` and then get it. I found a method to set in https://github.com/stefanbirkner/system-lambda, but it may be too complex.

Reply:
There's a `org.apache.flink.core.testutils.CommonTestUtils` from flink testutils, which could set the environment for testing purpose.   You may follow the demo [here](https://github.com/apache/flink/blob/dffa19b85bbe16946c960d9dfe0a2c2f26a0757b/flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopConfigLoadingTest.java#L253).

Reply:
Thanks, I will check it out.

Reply:
BTW,  I'd prefer to split this `mergeHiveConf` into three code blocks,  rather than put them into the complex if-else-if .  That makes the code more clear: 

```java
  private static Configuration mergeHiveConf(Configuration hadoopConf, String hiveConfDir) {
    Configuration newConf = new Configuration(hadoopConf);

    // Approach : Parse the hive-site.xml from hive-conf-dir if possible.
    if (!Strings.isNullOrEmpty(hiveConfDir)) {
      Preconditions.checkState(Files.exists(Paths.get(hiveConfDir, "hive-site.xml")),
          "There should be a hive-site.xml file under the directory %s", hiveConfDir);
      newConf.addResource(new Path(hiveConfDir, "hive-site.xml"));
      return newConf;
    }

    // Approach 2: Parse the hive-site.xml from HIVE_HOME system environment variable.
    String hiveHome = System.getenv("HIVE_HOME");
    if (Files.exists(Paths.get(hiveHome, "conf/hive-site.xml"))) {
      newConf.addResource(new Path(hiveHome, "conf/hive-site.xml"));
      return newConf;
    }

    // Approach 3: Parse the hive-site.xml from classpath
    URL configFile = CatalogLoader.class.getClassLoader().getResource("hive-site.xml");
    if (configFile != null) {
      newConf.addResource(configFile);
    }

    return newConf;
  }
```

Reply:
Thanks, I have updated the code. Please helps review code again when available.